src/js/components/ContainerSettingsComponent.jsx

```diff
@@ -0,0 +1,265 @@
+        volumes: [true]
+      }
+    };
+  },
+
+  handleAddRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {$push: [true]}}
+    );
+    this.setState({rows: rows});
+  },
+
+  handleRemoveRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {[position]: {$set: false}}}
+    );
+    if (rows[rowKey].filter((exists) => exists).length === 0) {
+      rows[rowKey].push(true);
+    }
+    this.setState({rows: rows});
+  },
+
+  getPortMappingRow: function (i = 0) {
+    return (
+      <div key={`pm-${i}`} className="row duplicable-row">
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].containerPort`}
+            label="Container Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].hostPort`}
+            label="Host Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].servicePort`}
+            label="Service Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].protocol`}
+            label="Protocol"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <select>
+              <option value="" disabled="disabled" selected="selected">Select</option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}
+              </option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}
+              </option>
+            </select>
+          </FormGroupComponent>
+          <DuplicableRowControls
+            handleAddRow={this.handleAddRow.bind(null, "portMappings", i)}
+            handleRemoveRow={this.handleRemoveRow.bind(null, "portMappings", i)} />
```

Comment:
Yeah we could create a closure that we pass as a handler. We could also store the values as properties on `DuplicableRowControls` and then we have a handler on that component that will pass it to the `handleAddRow` callback. Similar to this:
```
// DuplicableRowControls
  # handleAddRow: function() {
    event.target.blur();
    event.preventDefault();
    this.props.onAddRow(this.props.rowKey, this.props.position);
  }
```

We'd probably want to name the callback `onAddRow`, as it would no longer be getting called directly from an event.